examples/query/react/basic/src/test/test-utils.tsx

```diff
@@ -0,0 +1,54 @@
+import { render as rtlRender } from '@testing-library/react'
+import type { RenderOptions } from '@testing-library/react'
+import React, { PropsWithChildren } from 'react'
+import { Provider } from 'react-redux'
+import { DeepPartial } from 'redux'
+import { setupStore } from '../store'
+import type { AppStore, RootState } from '../store'
+
+// in ./src/store.ts I have a comment asking if we can make the initialState
+// parameter be DeepPartial<RootState> as opposed to just RootState
+function setupStoreWithPartialState(
+  initialPartialState: DeepPartial<RootState>
+) {
+  const state = { ...initialPartialState, ...setupStore().getState() }
+  return setupStore(state)
+}
+
+// This type interface extends the default options for render from RTL, as well
+// as allows the user to specify other things such as initialState, store. For
+// future dependencies, such as wanting to test with react-router, you can extend
+// this interface to accept a path and route and use those in a <MemoryRouter />
+interface IWithStoreOptions extends Omit<RenderOptions, 'queries'> {
+  initialState?: DeepPartial<RootState>
+  store?: AppStore
+}
+
+function render(
+  ui: React.ReactElement,
+  {
+    initialState = {},
+    store = setupStoreWithPartialState(initialState),
+    ...renderOptions
+  }: IWithStoreOptions = {}
+) {
+  // For example, even though it's the basic project and doesn't use the components,
+  // if we wanted to go ahead and set up the Chakra stuff and MemoryRouter, etc,
+  // all we'd have to do is update the return in Wrapper to:
+  //
+  //  <Provider store={store}>
+  //    <ChakraProvider>
+  //      <BrowserRouter>
+  //        <App />
+  //      </BrowserRouter>
+  //    </ChakraProvider>
+  //  </Provider>
```

Comment:
Do you want me to go ahead and put in Chakra and React-Router as dependencies for the basic project and get them hooked up here for the render function?

Reply:
@msutkowski based on your comment here https://github.com/reduxjs/redux-toolkit/pull/1218#discussion_r659173421 it seems like I should go ahead and set it up with the Chakra provider and MemoryRouter, that way copying the files to the other example projects is easier and consistent?

Reply:
I would say to *not* put them here if it's not being used. The goal is just to have it set up so that adding the extra providers when they're required is next to no effort

Reply:
Sounds good, that makes sense.